doc/lag/Increasing LACP PDU timeout during warm-reboot.md

```diff
@@ -0,0 +1,119 @@
+the timeout could be increased by some amount (beyond the limits of the
+protocol), and after warm-reboot, the timeout would be restored to the normal
+value.
+
+## Requirements
+
+- Switch running a supported SONiC with patches in libteam for this feature on
+  both sides of the LAG
+
+## Assumptions
+
+Such a change is going against the LACP protocol, and as such, can only be
+supported if both sides of the LAG are running SONiC, and they are running a
+version of SONiC that understands this. If the peer side is not running a
+supported version of SONiC, or it is not running SONiC, then the current
+behavior is preseved.
+
+## Limitations
+
+There's a chance that the packet to change the retry count doesn't get to the
+peer device (due to congestion, for example). Because the protocol as defined
+below doesn't wait for an acknowledgment packet, there's no reliable way of
+knowing whether it was handled by the peer device or not.
+
+# Background
+
+LACP supports two rates for sending PDUs. There is a short rate, where a PDU is
+sent every 1 second, and a long rate, where a PDU is sent every 30 seconds. Both
+sides know what rate to expect from the other side. If 3 LACP PDUs are missed,
+then the LAG is considered to be down, and data traffic is stopped. This results
+in an effective timeout of 3 seconds when using the short rate and 90 seconds
+when using the long rate.
+
+# Protocol
+
+To change the number of retries, an Ethernet packet of the fillowing structure
+will be sent. This Ethernet packet will have an ethertype of 0x6300, and will
+not have an IPv4 or IPv6 layer on top of it. Instead, there will instead be
+multiple TLV fields, similar to LACP.
+
+The TLV types will be defined as follows:
+
+| Value | Description         |
+|-------|---------------------|
+| 0x01  | Actor Information   |
+| 0x02  | Partner Information |
+| 0x03  | Retry Count         |
+
+Both Actor Information and Partner Information have the following content:
+
+| Starting byte | Length | Description     |
+|---------------|--------|-----------------|
+|      0        |   2    | System Priority |
+|      2        |   6    | System ID       |
+|      8        |   2    | Key             |
+|     10        |   2    | Port Priority   |
+|     12        |   2    | Port            |
+|     14        |   2    | Padding         |
+
+Retry count have the following content:
+
+| Starting byte | Length | Description     |
+|---------------|--------|-----------------|
+|      0        |   2    | New retry count |
+|      2        |   2    | Padding         |
+
+When the retry count needs to be changed, the sending device must send a packet
+with ethertype 0x6300, and the data will contain the Actor Information, Partner
+Information, and Retry Count TLVs. The receiving device must validate the actor
+and partner information, and then update the retry count as specified. No
+acknowledgment packet is sent back.
```

Comment:
How would you ensure the packet reached the peer without ack?

Reply:
Ack will be added into the protocol.